presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -255,10 +254,10 @@ public Object evaluate()
         return visitor.process(expression, new NoPagePositionContext());
     }
 
-    public Object evaluate(int position, Page page)
+    public Object evaluate(SymbolResolver inputs)
     {
         checkState(!optimize, "evaluate(int, Page) not allowed for optimizer");
```

Comment:
At least the message needs to be changed. Though I don't know why optimize cannot be allowed here.

Reply:
I believe optimize is going to be deprecated as no code path supposed to touch that.

Reply:
It is just this method is only intended for test so it has no support for optimize.

Reply:
Actually, the reason it is named evaluate/optimize is because it is intended to be used with either optimize flag on or off. If optimize flag is on, we should always return evaluated value or throw an exception. If optimize flag is off (evaluate), we can return an expression if we cannot evaluate the expression.  

So I will leave it like this and fix the error message here.